docs/source/entities-advanced.mdx

```diff
@@ -0,0 +1,535 @@
+You achieve this with one of the following directives:
+
+* [`@shareable`](#using-shareable)
+* [`@provides`](#using-provides)
+
+Which directive you use depends on the following logic:
+
+```mermaid
+graph TB;
+  resolve("Can my subgraph always resolve this field,<br/>or only from certain query paths?");
+  resolve--Always-->shareable("@shareable");
+  resolve--"Only certain query paths"-->provides("@provides");
+```
+
+If you aren't sure whether your subgraph can always resolve a field, see [Using `@provides`](#using-provides) for an example of a subgraph that _can't_.
+
+> ⚠️ **Important:** If multiple subgraphs can resolve a field, **make sure each subgraph's resolver for that field behaves identically.** Otherwise, queries might return inconsistent results depending on which subgraph resolves the field.
+
+### Using `@shareable`
+
+The `@shareable` directive indicates that a particular field can be resolved by more than one subgraph:
+
+<CodeColumns>
+
+```graphql{4}
+# Products subgraph
+type Product @key(fields: "id") {
+  id: ID!
+  name: String! @shareable
+  price: Int
+}
+```
+
+```graphql{4}
+# Inventory subgraph
+type Product @key(fields: "id") {
+  id: ID!
+  name: String! @shareable
+  inStock: Boolean!
+}
+```
+
+</CodeColumns>
+
+In this example, both the Products and Inventory subgraphs can resolve `Product.name`. This increases the likelihood that a client's query can be resolved by fewer total subgraphs.
+
+> If a field is marked `@shareable` in _any_ subgraph, it must be marked `@shareable` or `@external` in _every_ subgraph that defines it. Otherwise, composition fails.
+
+
+### Using `@provides`
+
+The `@provides` directive indicates that a particular field can be resolved by a subgraph _at a particular query path_. Let's look at an example.
+
+Here, our Products subgraph defines a `Product.name` field and marks it `@shareable` (this means other subgraphs are allowed to resolve it):
+
+```graphql{4}
+# Products subgraph
+type Product @key(fields: "id") {
+  id: ID!
+  name: String! @shareable
+  price: Int
+}
+```
+
+Meanwhile, our Inventory subgraph can _also_ resolve a product's name, but _only_ when that product is part of an `InStockCount`:
+
+```graphql{4,10}:title=inventory
+# Inventory subgraph
+
+type InStockCount {
+  product: Product! @provides(fields: "name")
+  quantity: Int!
+}
+
+type Product @key(fields: "id") {
+  id: ID!
+  name: String! @external
+  inStock: Boolean!
+}
+```
+
+Here we're using two directives in combination: `@provides` and `@external`.
+
+* The `@provides` directive tells the gateway, "This subgraph can resolve the `name` of any `Product` object returned by `InStockCount.product`."
+* The `@external` directive tells the gateway, "This subgraph _can't_ resolve the `name` of a `Product` object, _except_ wherever indicated by `@provides`."
+
+> Note that to include a field in a `@provides` directive, that field must be marked as `@shareable` or `@external` in every subgraph that defines it.
```

Comment:
Maybe worth clarifying that if a field is in a `@provides`, then it must be `@external` in the subgraph having the `@provides`, but it must be `@shareable` in an subgraphs that resolve it?  That is, you cannot have a field that is both in a `@provides` and `@shareable` in the same subgraph and wondering if that sentence could lead one to think otherwise.